src/ipc/ipc3/helper.c

```diff
@@ -277,10 +277,14 @@ static void comp_specific_builder(struct sof_ipc_comp *comp,
 	case SOF_COMP_MODULE_ADAPTER:
 	case SOF_COMP_NONE:
 		config->process.type = proc->type;
-		config->process.size = proc->size;
 #if CONFIG_LIBRARY || UNIT_TEST
+		/* FIXME: protocol code paths shouldn't be different for testing */
+		config->process.size = proc->size;
 		config->process.data = proc->data + comp->ext_data_length;
 #else
+		/* Clamp size, don't trust the protocol-supplied value! */
+		config->process.size = MIN(proc->size,
+					   SOF_IPC_MSG_MAX_SIZE - sizeof(*proc));
```

Comment:
why subtract the sizeof(*proc) here, wouldn't we just want to clamp to max message size. TBH, we should just reject anything bigger than ```SOF_IPC_MSG_MAX_SIZE``` with an error.

Reply:
I think it's because `proc->size` is the size of the `.data[]` array at the end of `struct sof_ipc_comp_process` so the total size should fit, i.e. `sizeof(*proc) + proc->size <= SOF_IPC_MSG_MAX_SIZE`. But yes, making too large `.size` values an error would be good!

Reply:
Thanks for the reminder @lyakh - maybe data->size would be a better name.